src/Microsoft.Azure.SignalR/Startup/NegotiateMatcherPolicy.cs

```diff
@@ -66,7 +66,7 @@ private Endpoint CreateNegotiateEndpoint(RouteEndpoint routeEndpoint)
             // Replaces the negotiate endpoint with one that does the service redirect
             var routeEndpointBuilder = new RouteEndpointBuilder(async context =>
             {
-                await ServiceRouteHelper.RedirectToService(context, hubMetadata.HubType.Name, null);
+                await ServiceRouteHelper.RedirectToService(context, hubMetadata.HubType, null);
```

Comment:
@vicancy 
About your [comment ](https://github.com/Azure/azure-signalr/pull/1103/files#r522590282), `RedirectToService<T>` can't be used due to the limit here - only `Type` is available

Reply:
make `CreateNegotiateEndpoint` a generic method then? It only runs when start, while your approach creates the type for every request.

Reply:
The `CreateNegotiateEndpoint ` is used [here](https://github.com/Azure/azure-signalr/blob/e2fee6d10e41f5fc9a97e7a2c278a675cd2a2522/src/Microsoft.Azure.SignalR/Startup/NegotiateMatcherPolicy.cs#L52), there's still no way to get the `<THub>` here.

The `ApplyAsync` is invoked on *each negotiate request*, while `CreateNegotiateEndpoint` needs the `CandidateSet` as a input parameter. 

How about I change the `ServiceRouteHelper` to `ServiceRouteHelper<Thub>`, and DI to the gloabl service collection, and then get the service [here](https://github.com/Azure/azure-signalr/blob/e2fee6d10e41f5fc9a97e7a2c278a675cd2a2522/src/Microsoft.Azure.SignalR/Startup/NegotiateMatcherPolicy.cs#L69) and call the `ServiceRouteHelper<THub>.RedirectToService()` method? Then I don't need `INegotiateHandler` and `Type` of the hub any more.